client/src/webview/PreReqView.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+'use strict';
+import * as vscode from 'vscode';
+import * as ejs from 'ejs';
+import * as path from 'path';
+import { ExtensionUtil, ExtensionData, EXTENSION_DATA_KEY, DEFAULT_EXTENSION_DATA } from '../util/ExtensionUtil';
+import { View } from './View';
+import { Reporter } from '../util/Reporter';
+import { DependencyManager } from '../dependencies/DependencyManager';
+import { ExtensionCommands } from '../../ExtensionCommands';
+import { TemporaryCommandRegistry } from '../dependencies/TemporaryCommandRegistry';
+import { VSCodeBlockchainOutputAdapter } from '../logging/VSCodeBlockchainOutputAdapter';
+import { LogType } from '../logging/OutputAdapter';
+import { SettingConfigurations } from '../../SettingConfigurations';
+
+export class PreReqView extends View {
+
+    public dependencies: any;
+    public restoreCommandHijack: boolean = false;
+    public openHomePage: boolean = false;
+
+    constructor(context: vscode.ExtensionContext) {
+        super(context, 'preReq', 'Prerequisites');
+    }
+
+    async openPanelInner(panel: vscode.WebviewPanel): Promise<void> {
+        Reporter.instance().sendTelemetryEvent('openedView', {name: panel.title}); // Report that a user has opened a new panel
+
+        const extensionPath: string = ExtensionUtil.getExtensionPath();
+        const panelIcon: vscode.Uri = vscode.Uri.file(path.join(extensionPath, 'resources', 'logo.svg'));
+
+        panel.iconPath = panelIcon;
+        panel.onDidDispose(async () => {
+            let isComplete: boolean = true;
+
+            if (!this.restoreCommandHijack) {
+                // If they close the panel by pressing 'x', we need to determine whether to restore command hijacking.
+                // This is because it's possible to open the prereq page even after having installed all the prereqs.
+                const dependencyManager: DependencyManager = DependencyManager.instance();
+                isComplete = await dependencyManager.hasPreReqsInstalled();
+            }
+
+            if (isComplete) {
+                 // We need to restore the command hijacking
+                const originalExtensionData: ExtensionData = this.context.globalState.get<ExtensionData>(EXTENSION_DATA_KEY, DEFAULT_EXTENSION_DATA);
```

Comment:
i don't think this code should be here, i think  you need a function that does this somewhere else